### PR TITLE
Add cookie options to fetchGlobalID

### PR DESCRIPTION
**Pull request by imnutz**

### CODE REVIEW

File: lib/plugins/globalid.js [hunk 1]
```diff
@@ -8,17 +8,27 @@ var invariant = require('../utils/misc').invariant
...
+  cookieOptions = cookieOptions || {}
+
   if (!result['global_id']) {
     return null
   }
-  cookie.setItem(cookieName, result['global_id'], 6000)
+  var path = cookieOptions.path || ''
+  var domain = cookieOptions.domain || ''
+  var secure = cookieOptions.secure || ''
+  var maxAge = cookieOptions.maxAge || 6000
```
Comment (tung-vu-td) [inline lib/plugins/globalid.js:20]: Not sure why we need to provide defaults for these value? The current logic should work fine with undefined.

File: lib/vendor/js-cookies.js [hunk 2]
```diff
@@ -84,7 +84,23 @@ var Cookies = {
           break
       }
     }
-    document.cookie = [ encode(sKey), '=', encode(sValue), sExpires, sDomain ? '; domain=' + sDomain : '', sPath ? '; path=' + sPath : '', bSecure ? '; secure' : '' ].join('')
+
+    var secureAndSameSite = ''
+    if (sameSite && sameSite.toUpperCase() === 'NONE') {
+      // if SameSite is set to None, we need to add Secure
+      // otherwise setting cookie doesn't work in some browsers
+      secureAndSameSite = '; Secure; SameSite=' + sameSite
```
Comment (tung-vu-td) [inline lib/vendor/js-cookies.js:92]: We're overriding Secure here to true if customer deliberately set bsecure to false. IMHO, we should either document this, or honor customer config secure = false if they set it.

Comment (imnutz) [reply lib/vendor/js-cookies.js:92]: I'll mention this in document

File: lib/plugins/globalid.js (hunk 1 above)
Comment (imnutz) [reply lib/plugins/globalid.js:20]: It is just a habit for setting default values. I think it is ok to leave it as is

**Comment by imnutz** [comment]
@ylzen Hi Yao, plz review this

**Pull request merged**